utilities/BuildUtilities.groovy

```diff
@@ -370,3 +371,98 @@ def createDatasets(String[] datasets, String options) {
+		}
+
+		fileLevelOverwrites = fileLevelPropertiesCache.get(buildFile)
+		if (fileLevelOverwrites != null) {
+			fileLevelProperty = fileLevelOverwrites.getProperty(prop)
+			if (fileLevelProperty!=null){
+				if (props.verbose) println("** File level overwrite found for $buildFile - $prop: $fileLevelProperty")
+				if (props.verbose)
+					return fileLevelProperty
+			}
+		}
+	}
+
+	//return DBB File Property or Global property
+	return props.getFileProperty(prop, buildFile)
+}
+
+/*
+ * getFileProperty - verifies if a properties file for the buildFile exists to overwrite properties 
+ */
+def getFileProperty(String prop, String buildFile){
+	// validate if build framework allows overwrites
+	if (props.allowFileLevelOverwrites && props.allowFileLevelOverwrites.toBoolean()){
+		if(!fileLevelPropertiesCache.containsKey(buildFile)){
+			loadFileLevelOverwrites(buildFile)
+		}
+
+		fileLevelOverwrites = fileLevelPropertiesCache.get(buildFile)
+		if (fileLevelOverwrites != null) {
+			fileLevelProperty = fileLevelOverwrites.getProperty(prop)
+			if (fileLevelProperty!=null){
+				if (props.verbose) println("** File level overwrite found for $buildFile - $prop: $fileLevelProperty")
+				// check if overwrite is allowed
+				if (props.whitelistFileLevelOverwrites.split(",").contains(prop))
+					return fileLevelProperty
+				else
+				if (props.verbose) println("*! Overwrite for $prop not allowed. Not on whitelist. See build property whitelistFileLevelOverwrites.")
+			}
+		}
+	}
+
+	//return DBB File Property or Global property
+	return props.getFileProperty(prop, buildFile)
+}
+
+/*
+ * groovy metaprogramming to return properties to language scripts
+ *  - verifies if a propertiesfile for the buildFile exists and overwrites settings
+ */
+static def propertyMissing(String name) {
+	return props.getProperty(name)
+}
+
+/*
+ * Load file level overwrites property files into hashmap 
+ */
+def loadFileLevelOverwrites(String buildFile){
+
+	String member = getFileNameWithoutExtension(buildFile)
+	String propertyFile = getAbsolutePath(props.application) + "/$props.propertiesFileLocation/${member}.properties"
+	File file = new File (propertyFile)
+	Properties fileProperties = new Properties()
+
+	if (file.exists()){
+		FileReader reader = new FileReader(file);
+		fileProperties.load(new FileInputStream(file))
+		if(props.verbose ) println("** Loaded file level property overwrites for $buildFile:")
+		if(props.verbose ) println("   $fileProperties")
+	} else
+	if(props.verbose ) println("** No file level property file found for $buildFile")
+
+	// Adding set to file level property cache, adds null if no file was found.
+	fileLevelPropertiesCache.put(buildFile,fileProperties)
+
+}
+
+/*
+ * get filename without file extension
+ */
+def getFileNameWithoutExtension(String buildFile){
+	int pos = buildFile.lastIndexOf(".");
+	int posFolder = buildFile.lastIndexOf("/")
+
+	if (pos > 0) {
+		return buildFile.substring(posFolder+1, pos);
+	}
```

Comment:
You are already working on the handle of the extension for multiple source code (COBOL, DBD, PSD, ..) with same file name.